src/containers/community/CommunityPosts.js

```diff
@@ -38,13 +40,24 @@ class CommunityPosts extends React.Component {
   }
 
   render () {
-    const { community, params: { id }, location: { query } } = this.props
+    let { community, params: { id }, location: { query }, dispatch } = this.props
     const { currentUser } = this.context
 
+    const requestToJoin = () => {
+      if (!currentUser) return dispatch(navigate(`/signup?next=/c/${community.slug}`))
```

Comment:
skip the auto-triggering if, once signed in, it turns out they're already a member